File: Easy/Java/2765. Longest Alternating Subarray.java
class Solution {
    public int alternatingSubarray(int[] nums) {
        int n = nums.length; 
        int alt = 0;

        /*
            Alternating pattern:
            x1 - x0 = 1
            x2 - x1 = -1
            x3 - x2 = 1
            Substitution:
            x1 - x0 = i % 2
        */
        
        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n && nums[i] + (j - i) % 2 == nums[j]; j++){
                alt = Math.max(alt, j - i + 1);
            }
        }

        return 1 < alt ? alt : -1;
    }
}

File: Easy/Python/2765. Longest Alternating Subarray.py
class Solution(object):
    def alternatingSubarray(self, nums):
        n = len(nums)
        alt = 0

        """
            Alternating pattern:
            x1 - x0 = 1
            x2 - x1 = -1
            x3 - x2 = 1
            Substitution:
            x1 - x0 = i % 2
        """

        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] + (j - i) % 2 == nums[j]:
                    alt = max(alt, j - i + 1)
                else:
                    break
        
        return alt if 1 < alt else -1
